### PR TITLE
Styleguide Update

### PR DESCRIPTION
**Pull request by Mab879**
#### Description:
* XCCDF 1.2 Corrections
* Add CMake rules
#### Rationale:
* Ensure we are following the XCCDF 1.2 specification
* CMake didn't have any rules before

### CODE REVIEW

File: docs/manual/developer/04_style_guide.md [hunk 1]
```diff
@@ -382,9 +389,10 @@ Shall be true if the system needs to be rebooted in order for the changes to tak
 
 #### `strategy`
 
-Should be one of the following values:
+Shall be one of the following values:
 
 * configure
+* combination
```
Comment (marcusburghardt) [inline docs/manual/developer/04_style_guide.md:395]: Could you give an example of `combination`, please?

File: docs/manual/developer/04_style_guide.md [hunk 2]
```diff
@@ -248,10 +248,14 @@ Rules sections must be in the following order, if they are present.
     * All subsections are HTML-Like
     * If defined must have at least one of the following sub-sections:
         * `general`
-        * `dependency`
+        * `functionality`
```
[the remark this answers is not in the document]
Comment (Mab879) [reply docs/manual/developer/04_style_guide.md:251]: `regulatory`, `hardware`, and `dependency` are all used. But this was also done to align with the XCCDF spec.

File: docs/manual/developer/04_style_guide.md (hunk 1 above)
Comment (Mab879) [reply docs/manual/developer/04_style_guide.md:395]: I don't have an example off the top of my head. This was done to align the XCCDF 1.2 spec.

File: docs/manual/developer/04_style_guide.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply docs/manual/developer/04_style_guide.md:251]: Ah I see. I quickly checked https://csrc.nist.gov/schema/xccdf/1.2/xccdf_1.2.xsd and they are all there.
Thanks

**Pull request merged**